### DISCUSSION TITLE
Error in post-processing with pyfdstools

### DISCUSSION THREAD

**Discussion by ahsadeghi** [firemodels/fds#12687] [Q&A]
Dear @johodges 
I am using pyfdstools for plotting temperature at a line on a slice I had in my simulations. I have the following slice for temperature in the input file

`&SLCF QUANTITY='TEMPERATURE', CELL_CENTERED=.TRUE., PBY=0.0375/
`

The simulation has run for 25 seconds, and I use the following code for pyfdstools (with the correct path for resultDir  and outputDir)

```
import pyfdstools as fds

qty = "TEMPERATURE"

chid = "Model_small"
resultDir = "C:\\Users\\......\\" 
outputDir =  "C:\\Users\\.......\\"

time = 20.0
dt = None

axis = 2# axis (x = 1, y = 2, z = 3)
value = 0.0375 # location (axis position)
data = fds.query2dAxisValue(resultDir, chid, qty, axis, value, time=time, dt=dt, atol=1e-1)
fds.renderSliceCsvs(data, chid, outputDir)
x = data['x']
z = data['z']
data_slc = data['datas'][:,:,0]
fds.plotSlice(x, z, data_slc, axis)
```
and I get the following error

```
    data = fds.query2dAxisValue(resultDir, chid, qty, axis, value, time=time, dt=dt, atol=1e-1)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\k21140758\OneDrive - King's College London\Documents\ESFSS\pyfdstools\pyfdstools\extractPlot3Ddata.py", line 1191, in query2dAxisValue
    quantities, slcfFiles, dimensions, meshes, centers = readSLCFquantities(chid, resultDir)
                                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\k21140758\OneDrive - King's College London\Documents\ESFSS\pyfdstools\pyfdstools\extractPlot3Ddata.py", line 962, in readSLCFquantities
    centers.append(files['SLICES'][file.split(os.sep)[-1]]['CELL_CENTERED'])
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
TypeError: 'bool' object is not subscriptable
```

I wonder if you would please help me to find the problem. Thank you.

Regards,
Hosein

**Comment by johodges** [comment]
I'll take a look. Could you post your FDS input file?

**Reply by ahsadeghi** [reply]
Thank you. Please see below the input file.

[Model_small_valid_00075.fds.txt](https://github.com/firemodels/fds/files/14772533/Model_small_valid_00075.fds.txt)

I should note that I am doing a mesh independence study, and I didn't get this error in the cases with coarser meshes.

**Reply by johodges** [reply]
Are you using the deployed version from pypi or installed from source? The case works for me at time=0.05 on the latest master.

**Reply by ahsadeghi** [reply]
I followed the instruction on the GitHub page, and installed with pip. Is there another version?

**Reply by johodges** [reply]
There is not a fixed release schedule for the pypi repo. Option 2 and 3 in the README show how you can install from source instead which will allow you to get updates more quickly. However, I just pushed the latest master to pypi. If you upgrade with pip it should work with your case. Let me know if you run into issues and I'll track them down.

**Reply by ahsadeghi** [reply]
Thank you. I installed the latest version in both ways. I still get the same error. I also get the following error when I run the example 


```
Traceback (most recent call last):
  File "C:\Users\k21140758\OneDrive - King's College London\Documents\ESFSS\pyfdstools\pyfdstools\examples.py", line 509, in <module>
    runExamples()
  File "C:\Users\k21140758\OneDrive - King's College London\Documents\ESFSS\pyfdstools\pyfdstools\examples.py", line 494, in runExamples
    stretchedMeshExample()
  File "C:\Users\k21140758\OneDrive - King's College London\Documents\ESFSS\pyfdstools\pyfdstools\examples.py", line 280, in stretchedMeshExample
    grid, data, times = fds.readSLCF3Ddata(chid, resultDir, quantity, time=time, dt=dt)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\k21140758\AppData\Local\Programs\Python\Python311\Lib\site-packages\pyfdstools\extractPlot3Ddata.py", line 636, in readSLCF3Ddata
    d2 = scpi.interpn((xs, ys, zs), d, p, method='linear', fill_value=None, bounds_error=False)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\k21140758\AppData\Local\Programs\Python\Python311\Lib\site-packages\scipy\interpolate\_rgi.py", line 634, in interpn
    _check_dimensionality(grid, values)
  File "C:\Users\k21140758\AppData\Local\Programs\Python\Python311\Lib\site-packages\scipy\interpolate\_rgi.py", line 45, in _check_dimensionality
    raise ValueError("There are %d points and %d values in "
ValueError: There are 16 points and 46 values in dimension 0
```

I tested it for the case that had a coarse mesh, the following input file, and it works.

[Model_small_valid_003.fds.txt](https://github.com/firemodels/fds/files/14776477/Model_small_valid_003.fds.txt)

**Reply by johodges** [reply]
Interesting. The crash in the examples is expected because the stretched mesh example needs a bit of rework since I changed the interpolation routine. I'll comment that out for now and get it fixed when I have a bit of time. I'll run your fine case for a longer duration and see if it works for me. What version of FDS are you using?

**Reply by ahsadeghi** [reply]
thank you very much. It is FDS-6.8.0-1238. Do I need to update?

**Reply by johodges** [reply]
let's both run with the FDS 6.9.0 release.

**Reply by johodges** [reply]
I am able to process the fine results with the latest master as well as through a clean install with pip. Can you update your pyfdstools then  run the command below:

`python -c "import pyfdstools as fds; print(fds.__version__)"`

This should return 0.0.17 if you are pulling the right version of pyfdstools. If you get an error message that means you do not have the latest pyfdstools (I just added support for `__version__`).

**Reply by ahsadeghi** [reply]
Thank you very much. It is now working with FDS 6.9.0. Thanks for your help.

**Comment by johodges** [comment]
Answer was to update both FDS and pyfdstools to the latest versions so they were consistent. Some minor quality of life features were added in pyfdstools to help debugging in the future. Updated version of the software was packaged and published to pypi.
